src/gui/XournalView.cpp

```diff
@@ -86,40 +95,38 @@ void XournalView::staticLayoutPages(GtkWidget* widget, GtkAllocation* allocation
     xv->layoutPages();
 }
 
+
 auto XournalView::clearMemoryTimer(XournalView* widget) -> gboolean {
-    GList* list = nullptr;
+    widget->cleanupBufferCache();
+    return true;
+}
 
-    for (auto&& page: widget->viewPages) {
-        if (page->getLastVisibleTime() > 0) {
-            list = g_list_insert_sorted(list, page, reinterpret_cast<GCompareFunc>(pageViewIncreasingClockTime));
+auto XournalView::cleanupBufferCache() -> void {
+    const auto& [pagesLower, pagesUpper] = this->preloadPageBounds(this->currentPage, this->viewPages.size());
+    g_assert(pagesLower <= pagesUpper);
+
+    // Queue pages to remove, sorted by the oldest (i.e. smallest last visible time)
+    constexpr auto cmp = [](XojPageView* a, XojPageView* b) {
+        return a->getLastVisibleTime() < b->getLastVisibleTime();
+    };
+    std::vector<XojPageView*> toRemove;
+    // Keep in cache: current page + preload_before + preload_after + LRU capacity
+    for (size_t i = 0; i < this->viewPages.size(); i++) {
+        auto&& page = this->viewPages[i];
+        size_t pageNum = i + 1;
+        bool isPreload = pagesLower <= pageNum && pageNum <= pagesUpper;
+        if (page->getLastVisibleTime() > 0 && !isPreload) {
+            toRemove.insert(std::upper_bound(toRemove.begin(), toRemove.end(), page, cmp), page);
         }
     }
 
-    int pixel = 2884560;
-    int firstPages = 4;
-
-    int i = 0;
-
-    for (GList* l = g_list_last(list); l != nullptr; l = l->prev)  // older (higher time) to newer (lower time)
-    {
-        if (firstPages) {
-            firstPages--;
-        } else {
-            auto* v = static_cast<XojPageView*>(l->data);
-
-            if (pixel <= 0) {
-                v->deleteViewBuffer();
-            } else {
-                pixel -= v->getBufferPixels();
-            }
+    // Remove oldest pages exceeding (LRU cache capacity)
+    const size_t cacheCapacity = 0;
+    if (toRemove.size() > cacheCapacity) {
```

Comment:
This does not make sense. Either you have a cache or you don't but this is some kind of really awkward mix.

Reply:
This should've been named `extraCacheCapacity`, but I was going to remove this entire snippet anyways.